File: src/algorithm_java/leetcode/Problem11.java
package algorithm.leetcode;

public class Problem11 {
    public static int maxArea(int[] height) {
        if (height.length < 2) return 0;
        int i = 0, j = height.length - 1;
        int maxArea = 0;
        while (i < j) {
            int area = Math.min(height[i], height[j]) * (j - i);
            if(area > maxArea) maxArea = area;
            if(height[i] < height[j])
                i++;
            else
                j--;
        }
        return maxArea;
    }

    public static void main(String[] args) {
        System.out.println(maxArea(new int[]{1,1}));
    }
}

File: src/algorithm_cpp/problem11.cpp
#include<iostream>
#include<vector>
#include<algorithm>

using namespace std;

class Solution {
public:
    int maxArea(vector<int>& height) {
	int l = 0, r = height.size() - 1;
	int max = 0;
	while(r > l) {
	    int area = min(height[l], height[r]) * (r - l);
	    if (area > max) {
		max = area;
	    }
	    if (height[l] > height[r]) {
		r--;
	    } else {
		l++;
	    }
	}
	return max;
    }
};
